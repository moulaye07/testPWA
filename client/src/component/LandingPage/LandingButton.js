// eslint-disable-next-line no-unused-vars
import React, { Component } from 'react';
import { getProfile } from '../../store/actions/profileAction';
import { isLoggedIn } from '../../store/actions/loginActions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


export class LandingButton extends Component {

  componentDidMount() {
    if (localStorage.getItem('user') != null) {
      this.props.getProfile();
    }

    if (!this.props.profile[0] && this.props.login.isLoggedIn) {
      this.props.getProfile();
    }
  }

  render() {
    const isAuthenticated = this.props.login.isLoggedIn;
    const profileName = this.props.profile[0] && this.props.profile[0].firstName;
    const authenticatedUser = (<p> Welcome {profileName}</p>);
    const guestUser = (
      <div className="acc">
        <a href="/Login"><p>Log in</p></a>
        <a href="/CreateAccount"><p>Create Account</p></a>
      </div>)

    return (
      <>
        { isAuthenticated ? authenticatedUser : guestUser}
      </>
    );
  }
}

LandingButton.propTypes = {
  getProfile: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
  return {
    login: state.login,
    profile: state.profile.profile
  }
}

export default connect(
  mapStateToProps,
  { isLoggedIn, getProfile }
)(LandingButton);
