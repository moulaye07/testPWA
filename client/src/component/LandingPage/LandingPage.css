p,
h3 {
  font-family: "DM Serif Text", serif;
}

.landingPage {
  background-color: white;
  display: grid;
  grid-template-rows: 30% 10% 10% 20% 10% 10%;
  grid-auto-columns: auto;
  height: 100%;
  width: 100vw;
  align-items: center;
  text-align: center;
  margin-top: 3%;
  padding: 0;
}

a {
  display: none;
  height: 100%;
}

img.CircleRight {
  display: none;
}

img.LandingImg {
  background-color: white;
  margin: 10px auto auto;
  width: 500px;
}

h3 {
  font-size: 110%;
  display: none;
}

.acc {
  display: none;
  flex-direction: row;
  align-content: center;
  align-self: end;
}

.acc p {
  margin: auto;
  font-size: 200%;
  text-decoration: underline;
}

.acc p:hover {
  opacity: 0.5;
}

.homeIcon {
  height: 100%;
  margin: auto;
}

.textIns {
  font-size: 30px;
  margin: 10% 5% 0;
  align-self: end;
  display: none;
  font-family: "Avenir Black", "Arial Black", Arial, sans-serif;
}

.acc a {
  margin: auto;
  font-size: 40px;
  text-decoration: underline;
  font-family: "Avenir Black", "Arial Black", Arial, sans-serif;
}

.notifSize {
  display: inline;
  font-size: 150%;
}

@media (max-width: 768px) {
  .landingPage {
    background-color: white;
    display: grid;
    grid-template-rows: 30% 10% 10% 20% 10% 10%;
    grid-auto-columns: auto;
    height: 100%;
    width: 100%;
    align-items: center;
    text-align: center;
    margin-top: 3%;
    padding: 0;
  }

  a {
    display: grid;
    height: 100%;
  }

  div.imgContainer {
    /* display: grid; */
    width: 100vw;
    height: 100%;
  }

  img.CircleRight {
    display: grid;
    height: 100%;
    display: flex;
    display: grid;
    margin: 10% auto;
    background-color: #3c4f57;
    border-radius: 50%;
  }

  img.CircleRight:hover {
    opacity: 0.5;
  }

  img.LandingImg {
    width: 100%;
    height: 132%;
    margin: auto;
    background-color: white;
  }

  h3 {
    font-size: 160%;
  }

  .acc {
    display: flex;
    flex-direction: row;
    align-content: center;
    align-self: end;
  }

  .acc p {
    margin: auto;
    font-size: 120%;
    text-decoration: underline;
  }

  .acc p:hover {
    opacity: 0.5;
  }

  .homeIcon {
    height: 100%;
    margin: auto;
  }

  .textIns {
    font-size: 100%;
    margin: 0 5% 0 5%;
    display: inline;
    align-self: end;
  }

  .acc a {
    margin: auto;
    font-size: 15px;
    text-decoration: underline;
    font-family: "Avenir Black", "Arial Black", Arial, sans-serif;
  }

  .notifSize {
    display: none;
  }
}
