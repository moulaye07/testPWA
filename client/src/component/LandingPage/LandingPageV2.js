/* eslint-disable no-unused-vars */
// eslint-disable-next-line no-unused-vars
import React from 'react';
// import MYtineraryLogo from '../img/MYtineraryLogo.png';
// import circledRight2 from '../img/circled-right-2-white.png';
import MYtineraryLogo from '../img/mytinerary_logo.png';
import circledRight2 from '../img/right_button.png';

import './LandingPage.css';
import { isLoggedIn } from '../../store/actions/loginActions';
import { getProfile } from '../../store/actions/profileAction';
import { connect } from 'react-redux';
import LandingButton from './LandingButton'
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';


class LandingPage extends React.Component {

  componentDidMount() {
    this.props.isLoggedIn()
  }

  render() {
    let isAuthenticated = this.props.login.isLoggedIn
    let profile = this.props.profile

    return (
      <div className="landingPage">
        <img className="LandingImg" src={MYtineraryLogo} alt="homeLogo"></img>
        <p className="textIns">Find your perfect trip, designed by insiders who knows and love their cities.</p>
        <h3>Start Browsing</h3>
        <NavLink to="/Cities">
          <div className='imgContainer'>
            <img className="CircleRight" src={circledRight2} alt="homeLogo"></img>
          </div>
        </NavLink>
        <p className="textIns">Want to build your own MYtinerary?</p>
        {/*<LandingButton key={profile.id} profile={profile} isAuthenticated={isAuthenticated} />*/}
        <p className='notifSize'>Mytinerary only Works on your phone, please adjust your screen size.</p>
      </div>
    )
  }
}


LandingPage.propTypes = {
  getProfile: PropTypes.func.isRequired,
  profile: PropTypes.array.isRequired
};

const mapStateToProps = (state) => {
  return {
    login: state.login,
    //profile: state.profile.profile
  }
}

export default connect(
  mapStateToProps,
  { isLoggedIn, getProfile }
)(LandingPage);