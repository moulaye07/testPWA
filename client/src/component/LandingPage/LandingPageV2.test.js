// eslint-disable-next-line no-unused-vars
import React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line no-unused-vars
import LandingPage from './LandingPageV2'

describe('Landing Page Component', () => {
  it('should render-without errors', () => {
    const component = shallow(<LandingPage />);
    console.log(component.debug());
    const wrapper = component.find('.LandingPage');
    // expect(wrapper.length).toBe(1)
    console.log(wrapper.length)
    expect(component).toMatchSnapshot();
    // expect(1).toBe(1);
  }) 
})
