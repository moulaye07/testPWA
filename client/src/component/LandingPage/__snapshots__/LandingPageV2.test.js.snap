// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Component should render-without errors 1`] = `
<div
  className="landingPage"
>
  <img
    alt="homeLogo"
    className="LandingImg"
    src="MYtineraryLogo.png"
  />
  <p
    className="textIns"
  >
    Find your perfect trip, designed by insiders who knows and love their cities.
  </p>
  <h3>
    Start Browsing
  </h3>
  <a
    href="/Cities"
  >
    <img
      alt="homeLogo"
      className="CircleRight"
      src="circled-right-2.png"
    />
  </a>
  <p
    className="textIns"
  >
    Want to build your own MYtinerary?
  </p>
  <div
    className="acc"
  >
    <a
      href="/Login"
    >
      <p>
        Log in
      </p>
    </a>
    <a
      href="/CreateAccount"
    >
      <p>
        Create Account
      </p>
    </a>
  </div>
</div>
`;
